mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -176,14 +176,14 @@ public MailboxId save(Mailbox mailbox) throws MailboxException {
     }
 
     private boolean trySave(Mailbox cassandraMailbox, CassandraId cassandraId) {
-        boolean isCreated = mailboxPathV2DAO.save(cassandraMailbox.generateAssociatedPath(), cassandraId).block();
-        if (isCreated) {
-            mailboxDAO.retrieveMailbox(cassandraId)
-                .blockOptional()
-                .ifPresent(mbx -> mailboxPathV2DAO.delete(mbx.generateAssociatedPath()).block());
-            mailboxDAO.save(cassandraMailbox).block();
-        }
-        return isCreated;
+        return mailboxPathV2DAO.save(cassandraMailbox.generateAssociatedPath(), cassandraId)
+            .filter(isCreated -> isCreated)
+            .flatMap(mailboxHasCreated -> mailboxDAO.retrieveMailbox(cassandraId)
+                .flatMap(mailbox -> mailboxPathV2DAO.delete(mailbox.generateAssociatedPath()))
+                .then(mailboxDAO.save(cassandraMailbox))
```

Comment:
This save SHOULD always be performed, even if `mailboxDAO.retrieveMailbox(cassandraId)` returned empty